### PR TITLE
Feature/docs uml

### PR DESCRIPTION
**Pull request by dishmop**
## Proposed changes

Added sequence diagrams for skills documentation. It uses the MkDocs mermaid plugin.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/weather-skills.md [hunk 1]
```diff
@@ -79,6 +79,34 @@ aea delete my_weather_station
...
+        activate Weather_AEA
+        
+        Weather_AEA->>Search: register_service
+        Client_AEA->>Search: search
+        Search->>Client_AEA: list_of_agents
+        Client_AEA->>Weather_AEA: call_for_proposal
+        Weather_AEA->>Client_AEA: propose
+        Client_AEA->>Weather_AEA: accept
+        Weather_AEA->>Client_AEA: match_accept
+        Client_AEA->>Weather_AEA: Inform funds transfered
```
Comment (DavidMinarsch) [inline docs/weather-skills.md]: Transfer funds `off chain` . The key being that the transfer is not covered in the example

File: docs/weather-skills.md [hunk 2]
```diff
@@ -241,7 +269,35 @@ aea delete my_weather_station
...
+        
+        Weather_AEA->>Search: register_service
+        Client_AEA->>Search: search
+        Search-->>Client_AEA: list_of_agents
+        Client_AEA->>Weather_AEA: call_for_proposal
+        Weather_AEA->>Client_AEA: propose
+        Client_AEA->>Weather_AEA: accept
+        Weather_AEA->>Client_AEA: match_accept
+        Client_AEA->>Blockchain: transfer_funds
+        Blockchain->>Weather_AEA: transfer_complete
```
Comment (DavidMinarsch) [inline docs/weather-skills.md]: see carpark ..

File: tests/test_packages/test_skills/test_weather_ledger.py [hunk 3]
```diff
@@ -19,125 +19,125 @@
...
+        process_two = subprocess.Popen([
+            sys.executable,
+            '-m',
+            'aea.cli',
+            "run"
+        ],
+            stdout=subprocess.PIPE,
+            env=os.environ.copy())
+
+        # check the gym run ends
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_weather_ledger.py]: should update comment

File: tests/test_packages/test_skills/test_weather_ledger.py [hunk 4]
```diff
@@ -19,125 +19,125 @@
...
+
+        time.sleep(20.0)
+        process_one.send_signal(signal.SIGINT)
+        process_one.wait(timeout=20)
+        process_two.send_signal(signal.SIGINT)
+        process_two.wait(timeout=20)
+
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills/test_weather_ledger.py]: Ideally, we would check the logs at some point to see if the transaction actually took place.

Comment (dishmop) [reply tests/test_packages/test_skills/test_weather_ledger.py]: Will work on this.

**Pull request merged**